me-md/me-md
# Set state in RiskFactors and begin rendering Symptoms

## What does this PR do?
This PR touches a lot of files, I'm sorry it's massive. I should've branched off, but here we are.
- [x] Refactored some of the hooks to be more SRP and have more specific naming, i.e. `[location, setLocation]` instead of `[state, setState]`.
- [x] Incorporate the height variable with styling in other existing components to make the design more responsive to various screen sizes.
- [x] Toggle, render and pass `present` state of risk factors to Symptoms component
- [x] Render symptoms upon search

### Where should the reviewer start?
In `RiskFactors.js` please look over the `findRiskToUpdate` method on line 42 and see if you can spot why after a second click, (no matter which risk factor is clicked) the `presentFactors` is logging a `true` or `false` depending on the state of the checkbox that is clicked. It does not happen on the very first click upon the screen loading. _I've attached a GIF of my console.logs to this PR_

Another thing I would like some feedback on is how much of the data cleaning logic we would like to implement in the component versus in a utils/helpers.js file? According to Evette, all we need to send in the evidence array is the id's of the risk factors that are "present".

Also, still considering moving the fetch to an earlier screen and passing down the risk factors as props because the fetch does take a while. Should these be pre-loaded, or a loading icon? Mull it over and let me know your thoughts.

In `SelectSymptoms.js`, I haven't gotten near as far as I have with the risk factors, but like I mentioned above, how much logic do we want in the component versus a helpers file for the Q&A fetch? Let me know what you think. Also, I know the styling is weird. I've been toying with it and it is very grumpy. Will revisit with a rested brain.
### How should this be manually tested?
Navigate to the Risk Factors screen and toggle the factors! (Also please throw in a console.log of the presentFactors state array once it has been updated to see what I mean about the additional true/falses that are being added. Could totally be a 2AM brain bug)

Also navigate to the Symptoms Screen and type in an area of the body, a symptom or condition you can think of! The toggling functionality is not yet in place.
### Have tests been written for all functionality?
No
### Any background context you want to provide?
I still have a lot of work to do on the symptoms, but before diving too far into it, I would love feedback on the state and data cleaning questions above.
#### What are the relevant tickets?
#6 
Closes #26 
Closes #27 
Closes #28 
Closes #29 
#30 
#32 
#### Screenshots (if appropriate)
<img width="400" alt="Risk Factors page" src="https://user-images.githubusercontent.com/31895658/71509813-3c297280-2852-11ea-8efc-74988746a6c8.png">
<img width="300" alt="console log statements with extra true/false strings in state" src="https://user-images.githubusercontent.com/31895658/71509833-506d6f80-2852-11ea-9f55-aaecd72effa9.gif">
<img width="335" alt="Select symptoms page" src="https://user-images.githubusercontent.com/31895658/71509882-91658400-2852-11ea-919b-08e67e5a4347.gif">

#### Additional Notes
I couldn't sleep, so here's a 2AM CST PR 😅
